conda_content_trust/common.py

```diff
@@ -332,56 +219,70 @@ class PrivateKey(
         value for sign() is a length 64 bytes() object, a raw ed25519
         signature.
     """
+    @classmethod
+    def to_bytes(cls, key):
+        return key.private_bytes(
+            encoding=serialization.Encoding.Raw,
+            format=serialization.PrivateFormat.Raw,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+
+    @classmethod
+    def from_bytes(cls, key_value_in_bytes):
+        """
+        Constructs an object of the class based on the given key value.
+        The "cryptography" library provides from_public_bytes() and
+        from_private_bytes() class methods for Ed25519PublicKey and
+        Ed25519PrivateKey classes in place of constructors.  We extend provide
+        a single API for those, and make the created objects objects of the
+        subclass using this mix-in.
+        """
+        # from_private_bytes() and from_public_bytes() both check length (32),
+        # but do not produce helpful errors if the argument provided it is not
+        # the right type, so we'll do that here before calling them.
+        checkformat_byteslike(key_value_in_bytes)
+        return super().from_private_bytes(key_value_in_bytes)
 
     def public_key(self):  # Overrides ed25519.Ed25519PrivateKey's method
         """
         Return the public key corresponding to this private key.
         """
-        # TODO: ✅❌⚠️💣  Confirm that this override works.  We MUST override
-        #                   the public_key() method.  If we just let the
-        #                   parent class's public_key() method be called, we'll
-        #                   get an object of the wrong type.
-        public = super().public_key()  # TODO: ✅ Python 2 compliance
-        public.__class__ = PublicKey  # TODO: ✅ This should not be hardcoded?
-
+        public = super().public_key()
         checkformat_key(public)
         return public
```

Comment:
Is this necessary anymore? `checkformat_key` is only doing a type check so unless `super().public_key()` returns different objects at different times this would be redundant to check

Reply:
It's one of the things I wanted to talk to Seb about, I'm not sure why the type check is done in the first place to be honest. Feels overly cautious, and I may just miss the reason?

Reply:
happy to leave it as is until we get more insight